### PR TITLE
[gnome] updates and fixes for gnome 47 and gcc 14

### PR DESCRIPTION
**Pull request by fabianhjr**
## Description of changes

Targetting `gnome` update branch/merge request: https://github.com/NixOS/nixpkgs/pull/333911

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ti/tinysparql/package.nix [hunk 1]
```diff
@@ -104,11 +104,7 @@ stdenv.mkDerivation (finalAttrs: {
     "-Dsystemd_user_services=false"
   ];
 
-  doCheck =
-    # https://gitlab.gnome.org/GNOME/tinysparql/-/issues/402
-    !stdenv.isDarwin
-    # https://gitlab.gnome.org/GNOME/tinysparql/-/issues/398
-    && !stdenv.is32bit;
+  doCheck = false;
```
Comment (fabianhjr) [inline pkgs/by-name/ti/tinysparql/package.nix:107]: oops shouldn't have included this

**Comment by bobby285271** [comment]
Thanks, usually I generate the commits in this way so it can contain the diff url like https://gitlab.gnome.org/GNOME/libdex/-/compare/0.7.1...0.8.0.

```bash
# Update everything that uses gnome-update-script
nix-shell maintainers/scripts/update.nix --arg predicate '(path: pkg: pkg.updateScript.name or null == "gnome-update-script")' --argstr keep-going true --argstr commit true --argstr max-workers 3

# Optional, for some reasons "→" is preferred in GNOME packages
export GIT_EDITOR="sed 's/->/→/' -i"
export GIT_SEQUENCE_EDITOR="sed 's/^pick /reword /' -i"
git rebase $base_commit -i
unset GIT_EDITOR
unset GIT_SEQUENCE_EDITOR
```

**Comment by fabianhjr** [comment]
Oh, was missing `--argstr commit true` when I ran the update script Dx

**Comment by honnip** [comment]
Can you revert https://github.com/NixOS/nixpkgs/pull/343127/commits/f97b3fe420708c621059ee61b21b1faa06944499? Then please add extensions for gnome47 to `gnomeExtensions` set:

https://github.com/NixOS/nixpkgs/blob/b7a0ca2c5b62ccab2b20b4ac148b4275d32b9647/pkgs/desktops/gnome/extensions/default.nix#L69

I will update and fetch gnome47 extensions on #335820

**Comment by fabianhjr** [comment]
Sure, rolling back

**Comment by fabianhjr** [comment]
> Can you revert https://github.com/NixOS/nixpkgs/commit/f97b3fe420708c621059ee61b21b1faa06944499? Then please add extensions for gnome47 to gnomeExtensions set:

Noticed this was edited, probably will be easier for you to add it to the mentioned merge request @honnip.

**Comment by bobby285271** [comment]
Merged:

8448f69573ce2ab99022ece6586c1e557a6c3cbf for g-s-d,
e2265d18dfa789fb7a03756b8bc3378598edb127 for gdm,
56df4b811ff37ab6e99538c6756b463a8d1610d2 for gnome-contacts,
36f62b29341f4b6c343cb9bb18b2c2336c939774 for glib,
db65c15447369966d2133dc3cb7c911b39a6be44 for at-spi2-core (also fixed diff url),
a3f88b33cad761f6f8845870ba76116f28991a49 for g-d-s,
79f566dd2e3ca6401442c059021e0dd23fe9730b for localsearch,
07bc1d0c06bf34e9371611823e41c98c342a1d39 for tinysparql (also fixed cli test in bd00ee6e1be2366dd99728b2d5eded73f3b2b966),
35ce371d6f523f1c4b6823ef0a8b8dbd2997d3ab for webkitgtk,
2a24dc85b5fbc70e69a8e651c598da4993c8a096 for gnome-session,
4909c379d12c0b237b50eb331da9e9c7241cb8a4 for seahorse,
07b01665427e66df5ec61892a3d2d9eda8340c29 for gnome-console (also fixed diff url),
734c2a77c44a95264272b0a4d45b028a9579834c for x-d-p-gnome,
d1171cb3880e2a024e45cbd1dd286858d1febb7d for orca (also fixed diff url and fixed rotten patch);

The gedit bump requires new tepl (with [api breaks](https://gitlab.gnome.org/World/gedit/libgedit-tepl/-/blob/6.11.0/docs/reference/api-breaks.xml#L313)) and other libgedit-* bumps, handled as bbf75a3c3e4fd7453ab14166eb631f9fbb345866 eade58d202e65fb8bf602bf5d4e2f74939999c2e e35907492cfa4f2630fb42c49fa28a566ee853fb f85803fd889665d3120c0950067a545c6a1c8fa6 ce8da7980164ba7ac7db37707475c775f768e256;
The GJS bump integrates [gobject-introspection-tests](https://gitlab.gnome.org/GNOME/gobject-introspection-tests) and need some fixups, initially handled as c5ff6dae0ae6b8d4bebdbc48f83fb83faed63383;
The GCC bump was [reverted](https://github.com/NixOS/nixpkgs/commit/41e40d8102d1ccbf717241fc259f25bd05ff596c) in staging-next, thus I would not look into 004a6422a0688b3f7594a1683aed285796debf84 and cf12629e3502dc772c2d128623ce4703dcfb0dc9 for now.

Thanks again! :heart:

**Pull request closed without merging**